lib/websocket-endpoint.js

```diff
@@ -147,7 +147,8 @@ class WebsocketEndpoint {
     drain() {
         var self = this;
 
-        if (self._ws.readyState !== WebSocket.OPEN) {
+        if (self._ws.readyState !== WebSocket.OPEN ||
+            self._message_queue.isEmpty()) {
```

Comment:
I think it should be discouraged and the current code discourages it. If there were a way to make drain() private I would.

If you're thinking of lines 165 where it does a check for the queue being empty before calling drain again, I'd say keep that check, if only to make the base case cleaner (to me anyway).